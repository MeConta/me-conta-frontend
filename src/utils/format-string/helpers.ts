// Rever tipagem
export const formatPhoneNumber = (value: string): any =>
  value
    .replace(/\D/g, '')
    .replace(/(\d{2})(\d)/, '($1) $2')
    .replace(/(\s\d{1})/, '$1 ')
    .replace(/(\d{4})(\d)/, '$1-$2')
    .replace(/(-\d{4})(\d+?)$/, '$1')

export const removePhoneMask = (value: string) => value.replace(/\D/g, '')
