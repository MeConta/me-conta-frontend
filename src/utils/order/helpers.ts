import { VolunteerAvailableSlot } from '../../services/volunteers-service/volunteer-service'

function convertSlotsIntoString(dateSlot: VolunteerAvailableSlot) {
  return new Date(dateSlot.inicio).toLocaleTimeString([], {
    hour: '2-digit',
    minute: '2-digit',
    hour12: false
  })
}

export const sortSlotsByTime = (dateSlot: VolunteerAvailableSlot[]) => {
  return dateSlot.sort((a, b) => {
    const x = convertSlotsIntoString(a)
    const y = convertSlotsIntoString(b)
    return x == y ? 0 : x > y ? 1 : -1
  })
}
