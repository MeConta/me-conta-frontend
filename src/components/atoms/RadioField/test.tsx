import { render, screen, waitFor } from '../../../utils/tests/helpers'
import { RadioField } from '.'
import userEvent from '@testing-library/user-event'

describe('<RadioField/>', () => {
  it('should not change value with text input', async () => {
    render(
      <RadioField
        name="name"
        options={['teste']}
        data-testid="radio"
        label="radio"
        readOnly
      />
    )
    expect(screen.getByTestId('radio')).toBeInTheDocument()
  })

  it('should render all options', async () => {
    const options = ['test-1', 'test-2', 'test-3']
    render(
      <RadioField
        name="name"
        options={options}
        data-testid="radio"
        label="radio"
        readOnly
      />
    )
    const elements = screen.getAllByTestId('radio')
    expect(elements.length).toBe(options.length)
  })

  it('should disable input with disabled propriety', async () => {
    const options = ['test-1', 'test-2', 'test-3']
    const mockChange = jest.fn()
    render(
      <RadioField
        name="name"
        data-testid="radio"
        options={options}
        disabled
        onChange={mockChange}
      />
    )
    const input = screen.getByDisplayValue(options[0])
    expect(input).toBeDisabled()

    userEvent.click(input)
    await waitFor(() => {
      expect(mockChange).not.toHaveBeenCalled()
    })
  })

  it('should render error the input', () => {
    const errorMessage = 'Error message'
    const options = ['test-1', 'test-2', 'test-3']
    const { container } = render(
      <RadioField
        name="name"
        data-testid="radio"
        options={options}
        error={errorMessage}
        readOnly
      />
    )
    expect(screen.getByText(errorMessage)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
