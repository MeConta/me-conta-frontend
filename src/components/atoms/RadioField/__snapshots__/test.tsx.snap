// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioField/> should render error the input 1`] = `
.c0 {
  margin-bottom: 1.6rem;
}

.c0 .c2 {
  border-color: #DE3163;
}

.c5 {
  font-size: 1.2rem;
  font-weight: 300;
  color: #848A8C;
  margin-left: 0.8rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  vertical-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.7rem;
  padding: 0 1.6rem;
}

.c1 {
  margin-top: 0.8rem;
  border: none;
}

.c4 {
  height: 2rem;
}

.c6 {
  color: #DE3163;
  font-size: 1rem;
}

<div
  class="c0"
>
  <fieldset
    class="c1"
    id="name"
  >
    <div
      class="c2 c3"
    >
      <input
        class="c4"
        data-testid="radio"
        name="name"
        readonly=""
        role="radio"
        type="radio"
        value="test-1"
      />
      <label
        class="c5"
        for="test-1"
      >
        test-1
      </label>
    </div>
    <div
      class="c2 c3"
    >
      <input
        class="c4"
        data-testid="radio"
        name="name"
        readonly=""
        role="radio"
        type="radio"
        value="test-2"
      />
      <label
        class="c5"
        for="test-2"
      >
        test-2
      </label>
    </div>
    <div
      class="c2 c3"
    >
      <input
        class="c4"
        data-testid="radio"
        name="name"
        readonly=""
        role="radio"
        type="radio"
        value="test-3"
      />
      <label
        class="c5"
        for="test-3"
      >
        test-3
      </label>
    </div>
  </fieldset>
  <p
    class="c6"
  >
     
    Error message
     
  </p>
</div>
`;
